refactored

LeanderK opened the pull request:
Hey,
i want to use your code as a library. Its not tested yet, i also have to add some features. Its more about the general direction, i think it is cleaner now.
I would like to hear your feed feedback! :)

LeanderK committed:
refactored library

LeanderK committed:
switched from if to switch statement

crsmoro commented:
hi!
i didn't expected someone to find this so quickly..
i didn't even finished the description yet.. just wanted to commit somewhere before something happened to my hd and lost everything (you know, never happens only if have something important lol)..
there's also some hard coded paths (among others things), it's far from ideal..
but thanks a lot, really!
I have graduation school first thing tomorrow morning, but i will look into it after..

LeanderK commented:
I didn't expect it too ;) No problem, i expect some bugs waiting to be ironed out :D

LeanderK committed:
fixed some bugs, now correctly pauses/resumes/flushes

LeanderK committed:
fixed bugs, wich caused exceptions when restarting a connections.
closes the soundline etc when the connections becomes inactive.
handles play/pause request.
flushes correctly.

LeanderK commented:
Hey, I've fixed some bugs, but did not test the main-class since u use it as a library.
the audio player now currently pauses and resumes, and i deleted the cracking when a connections finished. I should also perform way better, since the audio-thread sleeps when there is no data available/pauses etc.

LeanderK committed:
fixed some bugs, now volume is working somewhat correctly

LeanderK commented:
I had trouble with the volume, essentially the lower 3rd was way to quiet but then it got loud too quickly.
Can you test it if it's working?

crsmoro commented:
trying to get along with github.. used svn all my life..
when i was send my changes you sent.. now i am all lost..

crsmoro commented:
while i´m trying to commit what i did..
here some notes..

pointerVazio (pointerBlank) always going to be blank, i don't know why, ask spotify or don't since we are not supposed to be doing this with a closed library LOL
so i put into the initSPConfig...

the credentials file is not mandatory, its a internal control to facilitate next time you run already come logged in using the last.. no need to exit..
since it store a blob returned by the lib, i dont really know how to generate outside to use as start point.. (i know it's a combination of three 3 things, username, deviceId and password)
_i seed that you create a construction with the credentials, are you using that file for something?_

because the deviceId it's one of things you have to read on the credentials file i had to split the validation method.. i need the deviceId to initialize (you can't change it after) and then after you initialized you can logon

when you are not running standalone you can start without a login.. you can logon in the webpage after the startup..

it's a slight slower after to pass the events through a for (i liked the idea, don't worry, just have to discover why is that happened)

LeanderK committed:
fixed stupid bug

LeanderK commented:
learning git is worth it ;)
yeah i know the for iterator is not the fastest thing in the world, do you really notice it? I mean one can simply replace it with a for-loop if its noticeable.

I don't use the web-part, i also don't us the credentials file, i use it as spotify connect library. All the pieces for a credentials constructor were there, so i just added one ;)

So what i have not tested is the Main class and the Web-app part.

LeanderK committed:
added ability to mute the AudioPlayer

crsmoro commented:
did I get this right?
on the commit thing..

LeanderK commented:
I think so ;)

crsmoro commented:
:)
great job on the audio thread by the way... huge difference here..
i spent some time trying to reduce 100% use of CPU on the very first versions..
it really makes a difference on limited ARM architecture..
thread is not my strong..

LeanderK commented:
Thank you! Multi-threading is complicated ;)

crsmoro commented:
about the volume, all normal here.. (i don't get it the whole negative thing, but well...)
sometimes when i´m using the iOS app to control not on the same wifi i get some delay specially on volume (but nothing annoying, one, two seconds tops)..

about the mult-threading, since you seems to be an expert :P 
when you call other method on the EXACTLY same time (almost impossible) that pump events is calling cashs the vm (i thing it's because you can't have mult instance of it)..
ex.: click play, run spotifyLib.SpPlaybackPlay() and on that same time run spotifyLib.SpPumpEvents() ... crash occours..
any ideas? maybe synchronized?

LeanderK commented:
yeah i can write it quickly. Btw what does the spotifyLib.SpPumpEvents() do?

crsmoro commented:
its basically asks the lib if there's any new events..
simple test: dont start that thread and you will see that no event will be triggered ;)
